### DISCUSSION TITLE
Nextjs NProgress not working in v11

### DISCUSSION THREAD

**Discussion by Darren120** [vercel/next.js#29503] [Help]
Is this a bug? idk. 

Code for reproduce 

`Router.events.on("routeChangeStart", NProgress.start);
Router.events.on("routeChangeError", NProgress.done);
Router.events.on("routeChangeComplete", NProgress.done);
function MyApp({ Component, pageProps }: AppProps) {

  return (

    <RecoilRoot>
        <Loader/>
      
        <Component {...pageProps}/>
     
     </RecoilRoot>
  )
  
}`

**Comment by wwwdepot** [comment]
Router events has changed in v11. It's using React hook now.
https://nextjs.org/docs/api-reference/next/router#routerevents
https://nextjs.org/docs/upgrading#update-usage-of-routerevents

**Reply by Darren120** [reply]
didnt work.

**Reply by Darren120** [reply]
> Router events has changed in v11. It's using React hook now.
> 
> https://nextjs.org/docs/api-reference/next/router#routerevents
> 
> https://nextjs.org/docs/upgrading#update-usage-of-routerevents

It is NProgress interfering with something. I made my own loader. And that work.

**Comment by sanjaiyan-dev** [comment]
I think you can use below package.
For more info [visit here](https://www.npmjs.com/package/nextjs-progressbar)

_I am extremely sorry if I am wrong because I started to learn Next js recently and still learning it 😄_

```shell
npm i nextjs-progressbar
```
